src/bonsai/bonsai/bim/module/cost/operator.py

```diff
@@ -560,19 +569,57 @@ def poll(cls, context):
         return True
 
     def _execute(self, context):
-        cost_schedule = core.import_cost_schedule_csv(tool.Cost, self.filepath, self.is_schedule_of_rates)
-        core.add_csv_filepath(tool.Cost, self.filepath, self.is_schedule_of_rates, cost_schedule)
-        return {"FINISHED"}
-
+        
+        store_path = self.filepath
+        if self.use_relative_path:
+            store_path = tool.Ifc.get_uri(self.filepath, use_relative_path=True)
+        
+        resolved_path = Path(tool.Ifc.resolve_uri(self.filepath))
+        if not resolved_path.exists():
+            self.report({"ERROR"}, f"File does not exist: '{store_path}' (resolved to '{resolved_path}')")
+            return {"CANCELLED"}
+            
+        cost_schedule = core.import_cost_schedule_csv(tool.Cost, str(resolved_path), self.is_schedule_of_rates)
+        if cost_schedule:
+            core.add_csv_filepath(tool.Cost, store_path, self.is_schedule_of_rates, cost_schedule)
+            CostSchedulesData.is_loaded = False
+            
+            return {"FINISHED"}
+        return {"CANCELLED"}
+        
+    def draw(self, context):
+        row = self.layout.row()
+        row.prop(self, "is_schedule_of_rates")
+        row = self.layout.row()
+        row.prop(self, "use_relative_path")
 
 class RefreshCostScheduleCsv(bpy.types.Operator, tool.Ifc.Operator):
     bl_idname = "bim.refresh_cost_schedule_csv"
     bl_label = "Refresh Cost Schedule CSV"
+    bl_description = "Refresh cost schedule data from the associated CSV file"
     bl_options = {"REGISTER", "UNDO"}
 
     @classmethod
+    def poll(cls, context):
+        props = tool.Cost.get_cost_props()
+        if not props.active_cost_schedule_id:
+            cls.poll_message_set("No active cost schedule")
+            return False
+            
+        file_path = core.get_cost_schedule_csv_filepath(tool.Cost, props.active_cost_schedule_id)
```

Comment:
@falken10vdl i think here is better to call the tool directly so you can delete this core function